lib/models/channel_model.dart

```diff
@@ -41,9 +43,13 @@ class ChannelModel extends ChangeNotifier {
     return this;
   }
 
-  bool handleNotifications(Event notification) {
-    channel!.send('run', notification.toMap());
-    return true;
+  Future sendEvent(Event notification) {
+    final completer = Completer();
+    channel!
+        .send('run', notification.toMap())!
+        .receive("ok", (body) => completer.complete(body))
+        .receive("error", (error) => completer.complete(error));
```

Comment:
For the error, the completer should be completeError if i'm not mistaken
see : https://api.dart.dev/stable/2.10.5/dart-async/Completer/completeError.html

```suggestion
        .receive("error", (error) => completer.completeError(error));
```